packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -17,79 +17,508 @@ import {
+          <option value='' disabled>
+            {placeholder}
+          </option>
+        ) : null}
+
+        {options.map((option) => (
+          <option key={option.value} {...option}>
+            {option.label}
+          </option>
+        ))}
+      </SelectButton>
+
+      <SelectEndIcon disabled={disabled} />
+    </InputWithIcon>
+  );
+}) as PolymorphicForwardRefComponent<'div', NativeSelectProps>;
+
+// ----------------------------------------------------------------------------
+
+// This is a smaller, stricter subset of CustomSelectProps, because native
+// <select> only supports string values, and does not support icons or sublabels.
+type NativeSelectProps = SelectCommonProps & {
   /**
-   * Close handler that closes the dropdown.
+   * Selected option value.
+   *
+   * Must be a string, because it is passed as an attribute to the native <select>.
+   *
+   * Alternatively, pass `null` to reset the value.
    */
-  close: () => void;
+  value?: string | null;
   /**
-   * Indicates whether an item is selected.
+   * Callback invoked when the selected value changes.
    */
-  isSelected: boolean;
+  onChange?: (
+    value: string,
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => void;
+  /**
+   * Array of options that populates the select menu.
+   *
+   * The `value` property of each option must be a string.
+   */
+  options: Array<{ label: string; value: string; disabled?: boolean }>;
+  /**
+   * Default value that is selected on initial render. This is useful when you don't want to
+   * maintain your own state but still want to control the initial value.
+   *
+   * If not passed, the first option (or placeholder) will be automatically selected.
+   */
+  defaultValue?: string;
+  /**
+   * Placeholder for when no item is selected.
+   *
+   * Will be rendered as a disabled option at the top of the list, and automatically
+   * selected when no `value` or `defaultValue` is provided.
+   */
+  placeholder?: string;
+  /**
+   * Props to pass to the select element.
+   */
+  triggerProps?: Omit<React.ComponentPropsWithRef<'select'>, 'size'>;
+  required?: boolean;
+  multiple?: never;
 };
 
-export type SelectOption<T> = {
+type SelectCommonProps = {
   /**
-   * Label of the item used in dropdown list and when selected.
+   * Disables select.
+   * @default false
    */
-  label: string;
+  disabled?: boolean;
   /**
-   * Sublabel of the item shown below the label.
+   * Modify size of select.
    */
-  sublabel?: React.ReactNode;
+  size?: 'small' | 'large';
   /**
-   * Modify height of the item.
-   * Use 'large' when any of the select options have `sublabel`.
-   *
-   * Defaults to 'large' if `sublabel` provided, otherwise 'default'.
+   * Status of select.
    */
-  size?: 'default' | 'large';
+  status?: 'positive' | 'warning' | 'negative';
+};
+
+// ----------------------------------------------------------------------------
+
+const CustomSelect = React.forwardRef((props, forwardedRef) => {
```

Comment:
Does `CustomSelect` intentionally no longer have the generic type `<T>`?

Reply:
Yes. It is a private component so doesn't need the extra TS noise. The "public" part of it works the same as before.